README.md

```diff
@@ -1,33 +1,108 @@
-cd dex-operator
-oc new-project dex-operator
+* You will need operator-sdk 1.9.0+
+* OpenShift cluster
+* Github OAuth Application, for now, we will use github as id provider
+* To narrow the scope, you will want the OCP cluster where you run dex to be using signed certificates, as opposed to the default self-signed certificates. If we use self-signed certificates, of course, we have to distribute the CA to each cluster that try to access the dex service. Of cource, this should be supported in the final cut. You can follow the steps defined here to enable LetsEncrypt certificates: https://github.com/open-cluster-management/sre-tools/wiki/ACM---Day-1#add-an-acme-certificate
 
-# To use prebuilt image. Otherwise, you can reset these variables and build your own.
-export IMAGE_TAG_BASE=quay.io/cdoan/dex-operator
-export VERSION=0.0.2
+### Steps
 
-# this will deploy the operator bundle as a catalog source and deploy the operator from there.
-make sdk-run
+1. build and deploy the operator/controller
 
-# test the DexClient CR
-oc apply -f config/samples/auth_v1alpha1_dexclient.yaml
+    ```bash
+    git clone https://github.com/identitatem/dex-operator.git
+    cd dex-operator
+    oc new-project dex-operator
 
-# test the DexServer CR
-oc apply -f config/samples/auth_v1alpha1_dexserver.yaml
-```
+    # HACK - for now, manually create the clusterrole/binding to give dex access to dex resources
+    oc apply -f hack/community/rbac-dex-operator-dexsso-community.yaml
+
+    # Set the path to your quay repo.
+    export IMAGE_TAG_BASE=quay.io/cdoan/dex-operator
+    # OPTIONAL: arbitrarly change the version number of the image and bundle.
+    export VERSION=0.0.5
+
+    # build all the components
+    make bits
+
+    # deploy the operator from the bundle image (creates a catalog source). We can deploy the controller directly, but I started using the bundle and got used to it.
+    make sdk-run
+    ```
+
+2. verify the pods are running
+
+    ```bash
+    oc get pods
+    # At this point you should see this in the current namespace.
+    NAME                                                              READY   STATUS      RESTARTS   AGE
+    da3e6fd7599777c30bcf01817583a0dd66a5d8a2c09b7a802ea083b878lnf7x   0/1     Completed   0          25s
+    dex-operator-controller-manager-57b7464cb6-hr5d9                  2/2     Running     0          16s
+    quay-io-cdoan-dex-operator-bundle-v0-0-5                          1/1     Running     0          35s
+    ```
+
+3. if you have not done so already, generate or collect the github oauth information and set the appropriate environment variables. An example of a github app will have these values:
+
+    | key | value |
+    |-----|-------|
+    | Application name: | any-string |
+    | Homepage URL: | https://dex2-dex-operator.apps.pool-sno8x32sp2-w4qpg.demo.red-chesterfield.com |
+    | Authorization callback URL: | https://dex2-dex-operator.apps.pool-sno8x32sp2-w4qpg.demo.red-chesterfield.com/callback |
```

Comment:
Shall we make note that you don't actually know this value yet, and you can use a placeholder to create the GitHub app and come back and update it?

Reply:
I'll update in the next PR